src/androidTest/java/de/dennisguse/opentracks/io/file/exporter/ExportAsyncTaskTest.java

```diff
@@ -0,0 +1,353 @@
+package de.dennisguse.opentracks.io.file.exporter;
+
+import android.content.Context;
+import android.util.Pair;
+
+import androidx.documentfile.provider.DocumentFile;
+import androidx.test.core.app.ApplicationProvider;
+
+import org.junit.Assert;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.File;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Locale;
+
+import de.dennisguse.opentracks.content.data.TestDataUtil;
+import de.dennisguse.opentracks.content.data.Track;
+import de.dennisguse.opentracks.content.data.TrackPoint;
+import de.dennisguse.opentracks.content.provider.ContentProviderUtils;
+import de.dennisguse.opentracks.io.file.TrackFileFormat;
+import de.dennisguse.opentracks.stats.TrackStatistics;
+import de.dennisguse.opentracks.util.FileUtils;
+import de.dennisguse.opentracks.util.TrackNameUtils;
+
+public class ExportAsyncTaskTest {
+
+    private final Context context = ApplicationProvider.getApplicationContext();
+
+    private final ContentProviderUtils contentProviderUtils = new ContentProviderUtils(context);
+
+    private static final String TRACK_ICON = "the track icon";
+    private static final String TRACK_CATEGORY = "the category";
+    private static final String TRACK_DESCRIPTION = "the description";
+    private final long currentTimeMillis = System.currentTimeMillis();
+
+    private static final int NUM_TRACKS = 3;
+    private String[] trackStartTimes = new String[NUM_TRACKS];
+    private String[] trackEndTimes = new String[NUM_TRACKS];
+    private long[] trackIdList = new long[NUM_TRACKS];
+
+    static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'hh:mm:ss'Z'", Locale.US);
+
+    private File exportDirectoryFile;
+    private DocumentFile exportDirectory;
+
+    @Before
+    public void setUp() throws Exception {
+        // Begins with all empty.
+        contentProviderUtils.deleteAllTracks(context);
+
+        // Creates NUM_TRACKS start/end times.
+        loadStartEndTimes();
+
+        // Creates all tracks for the tests.
+        trackIdList = createAllTracks();
+
+        // Temporary directory work.
+        exportDirectoryFile = new File(context.getCacheDir().getAbsolutePath() + "/OpenTracks" + System.currentTimeMillis() + "/");
+        exportDirectoryFile.mkdir();
+        exportDirectory = DocumentFile.fromFile(exportDirectoryFile);
+
+        Assert.assertTrue(exportDirectoryFile.isDirectory());
+    }
+
+    @After
+    public void tearDown() {
+        for (int i = 0; i < trackIdList.length; i++) {
+            contentProviderUtils.deleteAllTracks(context);
```

Comment:
This should not happen in the for loop.
At least it looks weird.

Reply:
Ops, sorry. Fixed ;)